src/components/plugins/pagination/RemotePaginationHandler.js

```diff
@@ -0,0 +1,114 @@
+// @flow
+
+export default class RemotePaginationHandler {
+  page: Number;
+
+  totalRecords: Number;
+
+  pageSize: Number;
+
+  moveToNextPage: Function;
+
+  moveToPreviousPage: Function;
+
+  moveToFirstPage: Function;
+
+  moveToLastPage: Function;
+
+  getFirstRecordPosition: Function;
+
+  getLastRecordPosition: Function;
+
+  isOnFirstPage: Function;
+
+  isOnLastPage: Function;
+
+  getCurrentPage: Function;
+
+  getTotalNumberOfRecords: Function;
+
+  constructor(pageSize: number, totalRecords: number) {
+    this.page = 0;
+    this.pageSize = pageSize;
+    this.totalRecords = totalRecords;
+    this.moveToNextPage = this.moveToNextPage.bind(this);
+    this.moveToPreviousPage = this.moveToPreviousPage.bind(this);
+    this.moveToFirstPage = this.moveToFirstPage.bind(this);
+    this.moveToLastPage = this.moveToLastPage.bind(this);
+    this.getFirstRecordPosition = this.getFirstRecordPosition.bind(this);
+    this.getLastRecordPosition = this.getLastRecordPosition.bind(this);
+    this.isOnFirstPage = this.isOnFirstPage.bind(this);
+    this.isOnLastPage = this.isOnLastPage.bind(this);
+    this.getCurrentPage = this.getCurrentPage.bind(this);
+    this.getTotalNumberOfRecords = this.getTotalNumberOfRecords.bind(this);
+  }
+
+  getTotalNumberOfRecords() {
+    return this.totalRecords;
+  }
+
+  getTotalPages() {
+    this.totalPages = parseInt(this.totalRecords / this.pageSize, 10)
```

Comment:
Radix (10) is optional here so personally I would leave it out. I think everyone reading this would assume decimal anyway. Leaving it in forces readers to stop and understand what it represents.